src/schema/rules/datatypes/anat.yaml

```diff
@@ -27,6 +27,7 @@ nonparametric:
     reconstruction: optional
     run: optional
     part: optional
+    task: optional
```

Comment:
This is triggering a validation error of the schema because entities should be in the right order in those files.

https://github.com/bids-standard/bids-specification/actions/runs/3188511667/jobs/5201262576

I have tried to make the warnings informative enough so that it would tell you what the right order is.

https://github.com/bids-standard/bids-specification/actions/runs/3188511667/jobs/5201262576#step:5:86

Long story short: `task` comes before `acquisition

```python
  suffix group nonparametric in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'part', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'part']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group parametric in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'task', 'run']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group defacemask in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'modality', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'modality']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group multiecho in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'part', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'part']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group multiflip in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'flip', 'part', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'flip', 'part']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group multiinversion in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'inversion', 'part', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'inversion', 'part']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group mp2rage in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'flip', 'inversion', 'part', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'flip', 'inversion', 'part']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group vfamt in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'flip', 'mtransfer', 'part', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'echo', 'flip', 'mtransfer', 'part']
                  
    warnings.warn(

tools/schemacode/bidsschematools/tests/test_rules.py::test_entity_order
  /home/runner/work/bids-specification/bids-specification/tools/schemacode/bidsschematools/tests/test_rules.py:125: UserWarning: 
                  
  suffix group mtr in anat is out of order:
                  - got: ['subject', 'session', 'acquisition', 'ceagent', 'reconstruction', 'run', 'mtransfer', 'part', 'task']
                  - should be: ['subject', 'session', 'task', 'acquisition', 'ceagent', 'reconstruction', 'run', 'mtransfer', 'part']               
```

Reply:
Ok, but wasn't this correctly done in the .yaml files before?

Reply:
nope you had added task as the last entity in the list.

like this

```yaml
  entities:
    subject: required
    session: optional
    acquisition: optional
    ceagent: optional
    reconstruction: optional
    run: optional
    part: optional
    task: optional # not OK
```

and it should be like this


```yaml
  entities:
    subject: required
    session: optional
    task: optional # OK
    acquisition: optional
    ceagent: optional
    reconstruction: optional
    run: optional
    part: optional
```

make sense ?

Reply:
I'm not sure why we enforce this, btw. The ordering here is not meant to be authoritative, and this is stated in the docs.

Reply:
makes it easier for tools to know how to order entities in file names without having to refer to another part of the schema

Reply:
Ok, then let me just reorder things

Reply:
So the reordering is done for anat.